sql/core/src/test/scala/org/apache/spark/sql/sources/CreateTableAsSelectSuite.scala

```diff
@@ -212,7 +213,23 @@ class CreateTableAsSelectSuite extends DataSourceTest with SharedSQLContext with
       )
       val table = catalog.getTableMetadata(TableIdentifier("t"))
       assert(DDLUtils.getBucketSpecFromTableProperties(table) ==
-        Some(BucketSpec(5, Seq("a"), Seq("b"))))
+        Option(BucketSpec(5, Seq("a"), Seq("b"))))
+    }
+  }
+
+  test("create table using as select - with zero buckets") {
+    withTable("t") {
+      val e = intercept[AnalysisException] {
+        sql(
+          s"""
+             |CREATE TABLE t USING PARQUET
+             |OPTIONS (PATH '${path.toString}')
+             |CLUSTERED BY (a) SORTED BY (b) INTO 0 BUCKETS
```

Comment:
hm? so we allow `CLUSTERED BY` when schema is not specified for data source table?